src/query/functions/src/srfs/register.rs

```diff
@@ -0,0 +1,186 @@
+            DataType::Nullable(Box::new(DataType::Generic(0))),
+        ))))];
+        registry.register(
+            "unnest",
+            &arg_types,
+            &[DataType::Nullable(Box::new(DataType::Generic(0)))],
+            unnest_impl,
+        );
+    }
+
+    {
+        // Unnest 3d-array
+        let arg_types = vec![DataType::Array(Box::new(DataType::Array(Box::new(
+            DataType::Array(Box::new(DataType::Nullable(Box::new(DataType::Generic(0))))),
+        ))))];
+        registry.register(
+            "unnest",
+            &arg_types,
+            &[DataType::Nullable(Box::new(DataType::Generic(0)))],
+            unnest_impl,
+        );
+    }
+
+    {
+        // Unnest 4d-array
+        let arg_types = vec![DataType::Array(Box::new(DataType::Array(Box::new(
+            DataType::Array(Box::new(DataType::Array(Box::new(DataType::Nullable(
+                Box::new(DataType::Generic(0)),
+            ))))),
+        ))))];
+        registry.register(
+            "unnest",
+            &arg_types,
+            &[DataType::Nullable(Box::new(DataType::Generic(0)))],
+            unnest_impl,
+        );
+    }
+
+    {
+        // Unnest 5d-array
+        let arg_types = vec![DataType::Array(Box::new(DataType::Array(Box::new(
+            DataType::Array(Box::new(DataType::Array(Box::new(DataType::Array(
+                Box::new(DataType::Nullable(Box::new(DataType::Generic(0)))),
+            ))))),
+        ))))];
+        registry.register(
+            "unnest",
+            &arg_types,
+            &[DataType::Nullable(Box::new(DataType::Generic(0)))],
+            unnest_impl,
+        );
+    }
+
+    {
+        // Unnest 6d-array
+        let arg_types = vec![DataType::Array(Box::new(DataType::Array(Box::new(
+            DataType::Array(Box::new(DataType::Array(Box::new(DataType::Array(
+                Box::new(DataType::Array(Box::new(DataType::Nullable(Box::new(
+                    DataType::Generic(0),
+                ))))),
+            ))))),
+        ))))];
+        registry.register(
+            "unnest",
+            &arg_types,
+            &[DataType::Nullable(Box::new(DataType::Generic(0)))],
+            unnest_impl,
+        );
+    }
+
+    {
+        // Unnest 7d-array
+        let arg_types = vec![DataType::Array(Box::new(DataType::Array(Box::new(
+            DataType::Array(Box::new(DataType::Array(Box::new(DataType::Array(
+                Box::new(DataType::Array(Box::new(DataType::Array(Box::new(
+                    DataType::Nullable(Box::new(DataType::Generic(0))),
+                ))))),
+            ))))),
+        ))))];
+        registry.register(
+            "unnest",
+            &arg_types,
+            &[DataType::Nullable(Box::new(DataType::Generic(0)))],
+            unnest_impl,
+        );
+    }
+
+    {
+        // Unnest 9d-array
+        let arg_types = vec![DataType::Array(Box::new(DataType::Array(Box::new(
```

Comment:
8d is missing ?

Reply:
Why not making a common logic for any dimension?

Reply:
> 8d is missing ?

Fixed.